botocore/utils.py

```diff
@@ -48,6 +48,20 @@
 RETRYABLE_HTTP_ERRORS = (requests.Timeout, requests.ConnectionError)
 S3_ACCELERATE_WHITELIST = ['dualstack']
 
+CONTROLS = {
+    '\n': '\\n',
+    '\r': '\\r',
+    '\t': '\\t',
+}
+# Combines all CONTROLS keys into a big or regular expression
+ESCAPE_CONTROLS_RE = re.compile('|'.join(map(re.escape, CONTROLS)))
```

Comment:
Also it would be good to make these constants being with an underscore and only make `escape_controls` public. This just further restricts the interface we are exposing.